File: Inverted-Index/src/file_handler/Doc.java
package file_handler;

import java.util.List;

public class Doc {

    private String docName;
    private List<String> words;

    public Doc(String docName, List<String> words) {
        this.docName = docName;
        this.words = words;
    }

    public List<String> getWords() {
        return words;
    }
    public String getName() {
        return docName;
    }

    @Override
    public boolean equals(Object obj) {
        if(obj != null && obj.getClass().equals(this.getClass())) {
            final Doc doc = (Doc)obj;
            return this.docName.equals(doc.docName);
        }
        return false;
    }
    @Override
    public int hashCode() {
        return this.docName.hashCode();
    }

    @Override
    public String toString() {
        return String.format("%s", this.docName);
    }
}

File: Phase5-TDD/InvertedIndex/FileHandler/Doc.cs
using System;
using System.Collections.Generic;

namespace InvertedIndex.FileHandler
{
    public class Doc
    {
        private string _docName;
        private IList<string> _words;

        public Doc(string docName, IList<string> words)
        {
            this._docName = docName;
            this._words = words;
        }

        public IList<string> GetWords()
        {
            return _words;
        }

        public string GetName()
        {
            return _docName;
        }

        public override bool Equals(object obj)
        {
            
            if (obj == null || GetType() != obj.GetType())
            {
                return false;
            }
            Doc doc = (Doc)obj;
            return this._docName.Equals(doc._docName);
        }
        
        public override int GetHashCode()
        {
            return this._docName.GetHashCode();
        }

        public override string ToString()
        {
            return $"{this._docName}";
        }
    }
}